src/main/java/com/team4/testingsystem/services/impl/ResourceStorageServiceImpl.java

```diff
@@ -21,24 +19,34 @@ public ResourceStorageServiceImpl(FilesService filesService) {
     }
 
     @Override
-    public String upload(Resource file) throws FileSavingFailedException {
-        String fileUrl = UUID.randomUUID() + "-" + file.getFilename();
-        save(fileUrl, file);
-        return fileUrl;
+    public String upload(Resource file, Modules module, Long primaryId) {
```

Comment:
testId?

Reply:
That's not always testId, e.g. content file doesn't have testId, so primaryId is creator's id